kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/storage/AwsS3DirectoryLister.scala

```diff
@@ -20,94 +20,79 @@ import cats.implicits._
 import com.typesafe.scalalogging.LazyLogging
 import io.lenses.streamreactor.connect.cloud.common.config.ConnectorTaskId
 import io.lenses.streamreactor.connect.cloud.common.model.location.CloudLocation
-import io.lenses.streamreactor.connect.cloud.common.storage.DirectoryFindCompletionConfig
-import io.lenses.streamreactor.connect.cloud.common.storage.DirectoryFindResults
+import io.lenses.streamreactor.connect.cloud.common.storage.DirectoryLister
+import software.amazon.awssdk.services.s3.S3Client
 import software.amazon.awssdk.services.s3.model._
 
 import scala.jdk.CollectionConverters.IteratorHasAsScala
 
-object AwsS3DirectoryLister extends LazyLogging {
+class AwsS3DirectoryLister(connectorTaskId: ConnectorTaskId, s3Client: S3Client)
+    extends LazyLogging
+    with DirectoryLister {
+
+  private val listObjectsF: ListObjectsV2Request => IO[Iterator[ListObjectsV2Response]] = e =>
+    IO(s3Client.listObjectsV2Paginator(e).iterator().asScala)
 
   /**
     * @param wildcardExcludes allows ignoring paths containing certain strings.  Mainly it is used to prevent us from reading anything inside the .indexes key prefix, as these should be ignored by the source.
     */
-  def findDirectories(
+  override def findDirectories(
     bucketAndPrefix:  CloudLocation,
-    completionConfig: DirectoryFindCompletionConfig,
+    recurseLevels:    Int,
     exclude:          Set[String],
     wildcardExcludes: Set[String],
-    listObjectsF:     ListObjectsV2Request => Iterator[ListObjectsV2Response],
-    connectorTaskId:  ConnectorTaskId,
-  ): IO[DirectoryFindResults] =
+  ): IO[Set[String]] =
     for {
-      iterator <- IO(listObjectsF(createListObjectsRequest(bucketAndPrefix)))
-      prefixInfo <- extractPrefixesFromResponse(iterator,
-                                                exclude,
-                                                wildcardExcludes,
-                                                connectorTaskId,
-                                                completionConfig.levelsToRecurse,
-      )
-      flattened <- flattenPrefixes(
-        bucketAndPrefix,
-        prefixInfo.partitions,
-        completionConfig,
-        exclude,
-        wildcardExcludes,
-        listObjectsF,
-        connectorTaskId,
-      )
-    } yield DirectoryFindResults(flattened)
+      iterator   <- listObjects(bucketAndPrefix)
+      prefixInfo <- extractPrefixesFromResponse(iterator, exclude, wildcardExcludes, recurseLevels)
+      flattened  <- flattenPrefixes(bucketAndPrefix, prefixInfo, recurseLevels, exclude, wildcardExcludes)
+    } yield flattened
+
+  private def listObjects(bucketAndPrefix: CloudLocation): IO[Iterator[ListObjectsV2Response]] = {
+
+    def createListObjectsRequest(
+      bucketAndPrefix: CloudLocation,
+    ): ListObjectsV2Request = {
+
+      val builder = ListObjectsV2Request
+        .builder()
+        .maxKeys(1000)
```

Comment:
I thought the connector BATCH was controlling the number of keys. If so we should not hardcode the value